src/AppInstallerCLICore/Workflows/InstallFlow.cpp

```diff
@@ -364,11 +377,22 @@ namespace AppInstaller::CLI::Workflow
             }
             context << PortableInstall;
             break;
+        case InstallerTypeEnum::Zip:
+            context << ArchiveInstall;
+            break;
         default:
             THROW_HR(HRESULT_FROM_WIN32(ERROR_NOT_SUPPORTED));
         }
     }
 
+    void ArchiveInstall(Execution::Context& context)
+    {
+        context <<
+            ExtractInstallerFromArchive <<
+            VerifyAndSetNestedInstaller <<
+            ExecuteInstaller;

```

Comment:
My gut is to create `ExecuteInstallerForType(InstallerType)` and then have `ExecuteInstaller` call it with `installer.InstallerType` and this call it with `installer.NestedInstallerType` rather than the extra state management.

Reply:
Added `ExecuteInstallerForType(Context, InstallerType)` to handle the install flow selection logic based on installerType